Source/App.js

```diff
@@ -1 +1,417 @@
+
+    //Enable lighting based on sun/moon positions
+    viewer.scene.globe.enableLighting = true;
+
+    //////////////////////////////////////////////////////////////////////////
+    // Loading Imagery
+    //////////////////////////////////////////////////////////////////////////
+
+    Cesium.BingMapsApi.defaultKey = 'AsarFiDvISunWhi137V7l5Bu80baB73npU98oTyjqKOb7NbrkiuBPZfDxgXTrGtQ'; // For use on cesiumjs.org only
+
+    // Add Bing imagery
+    viewer.imageryLayers.addImageryProvider(new Cesium.BingMapsImageryProvider({
+        url : 'https://dev.virtualearth.net',
+        mapStyle: Cesium.BingMapsStyle.AERIAL // Can also use Cesium.BingMapsStyle.ROADS
+    }));
+
+    //////////////////////////////////////////////////////////////////////////
+    // Loading Terrain
+    //////////////////////////////////////////////////////////////////////////
+
+    //Load STK World Terrain
+    viewer.terrainProvider = new Cesium.CesiumTerrainProvider({
+        url : 'https://assets.agi.com/stk-terrain/world',
+        requestWaterMask : true, // required for water effects
+        requestVertexNormals : true // required for terrain lighting
+    });
+    //Enable depth testing so things behind the terrain disappear.
+    viewer.scene.globe.depthTestAgainstTerrain = true;
+
+    //////////////////////////////////////////////////////////////////////////
+    // Loading Data
+    //////////////////////////////////////////////////////////////////////////
+
+    var geojsonOptions = {
+        markerSize : 40,
+        markerSymbol : '!',
+        clampToGround : true
+    };
+    // Load points of interest from a GeoJson
+    var pointsPromise = Cesium.GeoJsonDataSource.load('./Source/SampleData/pointsOfInterest.geojson', geojsonOptions);
+
+
+    var kmlOptions = {
+        camera : viewer.scene.camera,
+        canvas : viewer.scene.canvas
+    };
+    // Load neighborhood boundaries from KML file
+    var neighborhoodsPromise = Cesium.KmlDataSource.load('./Source/SampleData/neighborhoods.kml', kmlOptions);
+
+    //////////////////////////////////////////////////////////////////////////
+    // Styling Data
+    //////////////////////////////////////////////////////////////////////////
+
+    var distanceDisplayCondition = new Cesium.DistanceDisplayCondition(10.0, 10000.0);
+    var neighborhoods = viewer.entities.add(new Cesium.Entity());
+    neighborhoodsPromise.then(function(dataSource) {
+        // Add the new data as entities to the viewer
+        viewer.dataSources.add(dataSource);
+
+        // Get the array of entities
+        var entities = dataSource.entities.values;
+
+        for (var i = 0; i < entities.length; i++) {
+            var entity = entities[i];
+            var name = entity.id;
+
+            if (Cesium.defined(entity.polygon)) {
+                //Set the polygon material to a random, translucent color.
+                entity.polygon.material = Cesium.Color.fromRandom({
```

Comment:
Ho ho ho! I just found a different dataset that's 2 mb smaller which runs much better. Thanks for the suggestion though! I'll try it on a bad machine.